### ISSUE TITLE
drop index throws SecurityAccessException, but deletes the index anyway

### ISSUE THREAD

**Issue by nagarajasr** [orientechnologies/orientdb#4693]
OrientDB 2.1-RC6

I tried to drop an index as a unprivileged user (having only reader and writer roles) and got a OSecurityAccessException, however the index was in fact deleted.

```sql

orientdb {db=indextestdb}> select from index:ouser.name

----+------+------+----
#   |@CLASS|key   |rid
----+------+------+----
0   |null  |admin |#5:0
1   |null  |nagu  |#5:3
2   |null  |reader|#5:1
3   |null  |writer|#5:2
----+------+------+----

4 item(s) found. Query executed in 0.038 sec(s).
orientdb> connect remote:localhost/indextestdb nagu infy1

Connecting to database [remote:localhost/indextestdb] with user 'nagu'...OK
orientdb {db=indextestdb}> drop index ouser.name


Removing index...
Error: com.orientechnologies.orient.core.exception.OSecurityAccessException: User 'nagu' has no the permission to execute the operation 'Update' against the resource: ResourceGeneric [name=CLUSTER, legacyName=database.cluster].internal

orientdb {db=indextestdb}> select from index:ouser.name

Error: com.orientechnologies.orient.core.exception.OCommandExecutionException: Target index 'OUSER.NAME' not found

orientdb {db=indextestdb}> select name from (select expand(roles) from ouser where name='nagu')

----+------+------
#   |@CLASS|name
----+------+------
0   |null  |writer
1   |null  |reader
----+------+------

2 item(s) found. Query executed in 0.022 sec(s).
orientdb {db=indextestdb}>

```

**Comment by nagarajasr** [comment]
The exception seems to be related to being unable to update the special target metadata:indexmanager, as the record still exists:

```sql

orientdb {db=indextestdb}> select expand(indexes) from metadata:indexmanager

----+------+------+--------+---------+---------+---------+---------+------------+------------------+----------------------------+------------------------------
#   |@CLASS|mapRid|clusters|algorithm|type     |name     |indexVers|valueContain|metadata          |indexDefinitionClass        |indexDefinition
----+------+------+--------+---------+---------+---------+---------+------------+------------------+----------------------------+------------------------------
0   |null  |#1:4  |[1]     |SBTREE   |UNIQUE   |OUser....|1        |NONE        |null              |com.orientechnologies.ori...|{className:OUser,field:name...
1   |null  |#1:0  |[0]     |SBTREE   |DICTIO...|dictio...|1        |NONE        |{durableInNonTx...|com.orientechnologies.ori...|{keyTypes:[1],collate:defau...
2   |null  |#1:2  |[1]     |SBTREE   |UNIQUE   |ORole....|1        |NONE        |null              |com.orientechnologies.ori...|{className:ORole,field:name...
----+------+------+--------+---------+---------+---------+---------+------------+------------------+----------------------------+------------------------------

3 item(s) found. Query executed in 0.033 sec(s).

```

**Comment by lvca** [comment]
This has been already fixed in the security fix pack or 2.1.1
